pkg/cmd/codespace/ssh.go

```diff
@@ -235,6 +237,65 @@ func shouldGenerateSSHKeys(args []string, opts sshOptions) bool {
 	return true
 }
 
+func setupAutomaticSSHKeys(sshContext ssh.Context) (*ssh.KeyPair, error) {
+	keyPair := checkAndUpdateOldKeyPair(sshContext)
+	if keyPair != nil {
+		return keyPair, nil
+	}
+
+	keyPair, err := sshContext.GenerateSSHKey(automaticPrivateKeyName, "")
+	if err != nil && !errors.Is(err, ssh.ErrKeyAlreadyExists) {
+		return nil, err
+	}
+
+	return keyPair, nil
+}
+
+// checkAndUpdateOldKeyPair handles backward compatibility with the old keypair names.
+// If the old public and private keys both exist they are renamed to the new name.
+// The return value is non-nil only if the rename happens.
+func checkAndUpdateOldKeyPair(sshContext ssh.Context) *ssh.KeyPair {
+	publicKeys, err := sshContext.LocalPublicKeys()
+	if err != nil {
+		return nil
+	}
+
+	for _, publicKey := range publicKeys {
+		if !strings.HasSuffix(publicKey, automaticPrivateKeyNameOld+".pub") {
+			continue
+		}
+
+		privateKey := strings.TrimSuffix(publicKey, ".pub")
+		_, err := os.Stat(privateKey)
+		if err != nil {
+			continue
+		}
+
+		// Both old public and private keys exist, rename them to the new name
+
+		publicKeyNew := strings.Replace(publicKey, automaticPrivateKeyNameOld, automaticPrivateKeyName, -1)
```

Comment:
`strings.Replace` isn't a safe way to rename a file because it will affect any string in the path component:

```go
strings.Replace("/home/mycodespacesuser/.ssh/codespaces.pub", "codespaces", "codespaces.auto", -1)
//=> "/home/mycodespaces.autouser/.ssh/codespaces.auto.pub"
```

I would suggest instead
```suggestion
		publicKeyNew := filepath.Join(filepath.Dir(publicKey), automaticPrivateKeyName+".pub")
```

Reply:
Good point! Updated to the suggestion